server/bin/gen-certs

```diff
@@ -47,19 +81,153 @@ if [ ! -d $UPSTREAM_CERTS_HOME ]; then
     $SUDO mkdir -p $UPSTREAM_CERTS_HOME
 fi
 
-HOSTNAME=${HOSTNAME:-$(hostname)}
 
 if [ -f $CA_KEY ] && [ -f $CA_CERT ] && [ "$FORCECERT" != "1" ]; then
     echo "Certificates are already present."
 else
+    echo
+    echo "Creating CA private key password"
+    sudo su -c "echo $RANDOM > $CA_KEY_PASSWORD"
```

Comment:
Could be, for CA_KEY_PASSWORD the script only uses it as input for 'openssl genrsa'.

The docs for that are a little vague, but sounds like openssl at least expects there can be a newline there.

```
       file:pathname
                 the first line of pathname is the password. If the same pathname argument is supplied to -passin
                 and -passout arguments then the first line will be used for the input password and the next line
                 for the output password. pathname need not refer to a regular file: it could for example refer to
                 a device or named pipe.
```